### ISSUE TITLE
Building xmlada component fails with GNAT 2019/How to control the compiler version

### ISSUE THREAD

**Issue by stcarrez** [alire-project/alire#265]
I'm using GNAT 2019 and when running:

```
alr get xmlada
alr build
```

It fails because the xmlada version that was downloaded is incompatible with GNAT 2019
and fails:

```
-> Loading index found at /home/ciceron/work/awa/alr-test/local-index/index/index.toml
-> Loading full catalog from /home/ciceron/work/awa/alr-test/local-index/index
-> Update completed
-> Using candidate project root: /home/ciceron/work/awa/alr-test/tst/xmlada_16.1.0_a2d45474
Compile
   [Ada]          input_sources-http.adb
input_sources-http.adb:194:07: missing argument for parameter "Level" in call to "Set_Socket_Option" declared at g-socket.ads:1283

   compilation of input_sources-http.adb failed

gprbuild: *** compilation phase failed
Warning: alr detected a compilation failure, re-run with -v or -d for details
ERROR: alr build unsuccessful
```

I guess this is because the xmlada definition in alire-index only refers to version 16.1 of XML/Ada.

Should we have an index per-compiler version?

A more general question is how can we say that a specific version of a package
requires a specific version of the compiler?

The issue comes when you want to make you package portable on several platforms
(Linux, Windows, FreeBSD, NetBSD, MacOS). We cannot guarantee to have the same
compiler on each of these targets.

**Comment by mosteo** [comment]
Currently this is supported through the `available` expressions in the [index](https://github.com/alire-project/alire/blob/master/doc/catalog-format-spec.rst#parameters). That's a bit unwieldy and I just realized the doc I linked to is outdated in regard to supported compilers (see #241). 

Among my next targets is to improve that among other related things. I am thinking that, at least for the compiler, the straightforward way is to include it as a dependency in crates that require specific versions. I will shortly open a experimental PR to discuss this very issue, so I can ping you from there if interested.

**Comment by mosteo** [comment]
This was implemented by making `gnat` available as a dependency and detection of the available compiler via externals (#295). A combination of `available` and `depends-on` definitions should be now flexible enough. Please reopen if you detect situations that cannot be expressed with the current syntax.

As for the concrete definitions of xmlada/gnatcoll crates, I think this will be fixed shortly by people from AdaCore.
